IoTCIntegration/lib/engine.js

```diff
@@ -75,14 +75,29 @@ function validateMeasurements(measurements) {
     }
 
     for (const field in measurements) {
-        if (typeof measurements[field] !== 'number' && typeof measurements[field] !== 'string') {
+        if (typeof measurements[field] !== 'number' && typeof measurements[field] !== 'string' && !isLocation(measurements[field])) {
             return false;
         }
     }
 
     return true;
 }
 
+/**
+ * @returns true if a measurement is a location.
+ */
+function isLocation(measurement) {
+    if (typeof measurement !== 'object' || typeof measurement.lat !== 'number' || typeof measurement.lon !== 'number') {
```

Comment:
might want to have a `measurement &&` in the front here since `typeof null === 'object'`

Reply:
Good catch. I really didn't know that :)